content/api/commands/mount.md

```diff
@@ -0,0 +1,607 @@
+  })
+
+  // Change location to `/login`
+  router.push('/login')
+
+  // Pass the already initialized router for use
+  cy.mountWithRouter(Navigation, { router: router })
+
+  cy.get('a').contains('Login').should('have.class', 'router-link-active')
+})
+```
+
+</template>
+<template #vue3>
+
+```js
+import { mount } from '@cypress/vue'
+import { createMemoryHistory, createRouter } from 'vue-router'
+import { routes } from '../../src/router'
+
+Cypress.Commands.add('mountWithRouter', (comp, options = {}) => {
+  // Setup options object
+  options.global = options.global || {}
+  options.global.plugins = options.global.plugins || []
+
+  // Use the router passed in via options,
+  // or the default one if not provided
+  options.router =
+    options.router ||
+    createRouter({
+      routes: routes,
+      history: createMemoryHistory(),
+    })
+
+  // Add router plugin
+  options.global.plugins.push({
+    install(app) {
+      app.use(options.router)
+    },
+  })
+
+  return mount(comp, options)
+})
+```
+
+Typings:
+
+```ts
+import { mount } from '@cypress/vue'
+import { Router } from 'vue-router'
+
+type MountParams = Parameters<typeof mount>
+type OptionsParam = MountParams[1] & { router?: Router }
+
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Helper mount function for Vue Components
+       * @param component Vue Component or JSX Element to mount
+       * @param options Options passed to Vue Test Utils
+       */
+      mountWithRouter(component: any, options?: OptionsParam): Chainable<any>
+    }
+  }
+}
+```
+
+Usage:
+
+Calling `router.push()` in the router for Vue 3 is an asynchronous operation.
+You can use the [cy.wrap](/api/commands/wrap) command to have Cypress await the
+promise's resolve before it continues with other commands:
+
+```js
+import Navigation from './Navigation.vue'
+import { routes } from '../router'
+import { createMemoryHistory, createRouter } from 'vue-router'
+
+it('login link should be active when url is "/login"', () => {
+  // Create a new router instance for each test
+  const router = createRouter({
+    routes: routes,
+    history: createMemoryHistory(),
+  })
+
+  // Change location to `/login`,
+  // and await on the promise with cy.wrap
+  cy.wrap(router.push('/login'))
+
+  // Pass the already initialized router for use
+  cy.mountWithRouter(<Navigation />, { router: router })
+
+  cy.get('a').contains('Login').should('have.class', 'router-link-active')
+})
+```
+
+</template>
+</code-group-vue2-vue3>
+
+### Global Components
+
+If you have components that are registered globally in the main application
+file, you will need to set them up in your mount command as well.
+
+<code-group-vue2-vue3>
+<template #vue2>
+
+```js
+import { mount } from '@cypress/vue'
+import Button from '../../src/components/Button.vue'
+
+Cypress.Commands.overwrite('mount', (comp, options = {}) => {
+  // Register global components
+  Vue.component('MyButton', Button)
```

Comment:
Is it a best practice to give the registered component the same or a different name?

```suggestion
  Vue.component('Button', Button)
```